tests/tasks/setup/kitctl/controlplane.yaml

```diff
@@ -68,8 +62,6 @@ spec:
         etcd:
           replicas: $(params.etcd-replicas)
           spec:
-            nodeSelector:
-              node.kubernetes.io/instance-type: $(params.etcd-instance-type)
```

Comment:
If we don't set this, how do make sure which instance the controlplane components will be running?

Reply:
Setting this here like this will make it run likely into karpenter unable to provision anything at all when there is icing. Instead there is a way to set this up through Karpenter-kit provisioner (or) you can use different bundle-types etc variables that karp can understand.

Reply:
But we don't change the Karpenter provisioner in our tests. If I need to create a guest cluster how do I define the instance type if we remove this?

Reply:
But you'd run into issues by hardcoding it to instance type as i did if you hardcode instance types like this. Ideally you'd want to take this approach - https://karpenter.sh/preview/tasks/scheduling/#supported-labels

Anyways all these changes are clean up and can be done as part of other PR and don't want to muddy the water from the original issue the PR is raised for. Because these changes were already committed to this branch many days back which is why these ended up in this PR. We can come back to these later may be like a separate git issue for validating the current kit pipelines w.rt current state of KIT repo.